meteor/server/api/playout/infinites.ts

```diff
@@ -274,12 +274,14 @@ export const cropInfinitesOnLayer = syncFunction(function cropInfinitesOnLayer (
 
 	let ps: Array<Promise<any>> = []
 	for (const piece of pieces) {
-		ps.push(asyncCollectionUpdate(Pieces, piece._id, { $set: {
-			userDuration: { end: `#${getPieceGroupId(newPiece)}.start + ${newPiece.adlibPreroll || 0}` },
-			definitelyEnded: getCurrentTime() + DEFINITELY_ENDED_FUTURE_DURATION + (newPiece.adlibPreroll || 0),
-			infiniteMode: PieceLifespan.Normal,
-			originalInfiniteMode: piece.originalInfiniteMode !== undefined ? piece.originalInfiniteMode : piece.infiniteMode
-		}}))
+		if (!piece.userDuration || !piece.userDuration.duration) {
+			ps.push(asyncCollectionUpdate(Pieces, piece._id, { $set: {
+				userDuration: { end: `#${getPieceGroupId(newPiece)}.start + ${newPiece.adlibPreroll || 0}` },
```

Comment:
I'm not entirely sure about all the `timelineObj.enable` mumbo-jumbo you guys do in the blueprints, but this seems like asking for trouble: if we check for `piece.userDuration.duration` and then overwrite `piece.userDuration` with `pieceUser.duration.end` (thus making sure it doesn't have a `duration`), won't this eventually get us in an infinite loop of sorts? Shouldn't we make sure that we only process these once? Shouldn't it be `(!piece.userDuration || (!piece.userDuration.duration && !piece.userDuration.end))`?

Reply:
Good catch, absolutely